cli/src/deploy.rs

```diff
@@ -53,14 +53,25 @@ pub async fn deploy(rpc_url: String, signer: Keypair, deploy_args: DeployArgs) -
                 access_key,
                 secret_key,
                 region,
+                endpoint,
                 ..
             } = s3_upload;
 
+            let dest =
+            object_store::path::Path::from(format!("{}-{}", manifest.name, manifest.image_id));
+
+            let url = if let Some(endpoint) = endpoint {
+                format!("{}", endpoint)
```

Comment:
Since `endpoint` is already a `String` here I believe you can just return that instead of creating a new `String` with the `format!` macro

Reply:
For even more style points you could try (didn't test it but should be close):
```rust
let url = endpoint.unwrap_or(format!("https://{}.s3.{}.amazonaws.com/{}", bucket, region, dest));
```

Reply:
Good catch! I’ve updated the code to return the existing endpoint directly instead of using format!. Thanks for pointing it out!